File: servers/dotnet/Domain/Blogs/Validators/BlogValidator.cs
using System;
using Domain.Blogs.DTOs;

namespace Domain.Blogs.Validators;

public class BlogValidator
{
    public Dictionary<string, string> Validate(BlogDto blogDto)
    {
        Dictionary<string, string> errors = new Dictionary<string, string>();

        if (string.IsNullOrWhiteSpace(blogDto.Title))
        {
            errors.Add("title", "Blog must contain a title");
        }
        if (string.IsNullOrWhiteSpace(blogDto.Body))
        {
            errors.Add("body", "Blog must contain a body");
        }

        return errors;
    }
}

File: servers/springboot/server/src/main/java/com/gabrielgavrilov/ohmyblog/core/blog/validator/BlogValidator.java
package com.gabrielgavrilov.ohmyblog.core.blog.validator;

import com.gabrielgavrilov.ohmyblog.api.BlogDto;
import org.apache.commons.lang3.StringUtils;
import org.springframework.stereotype.Component;

import java.util.HashMap;
import java.util.Map;

@Component
public class BlogValidator {
    public Map<String, String> validate(BlogDto blogDto) {
        Map<String, String> errors = new HashMap<>();

        if (StringUtils.isBlank(blogDto.getTitle())) {
            errors.put("title", "Blog title cannot be empty.");
        }

        return errors;
    }
}
